opm/simulators/linalg/bda/IncompleteSAI.cpp

```diff
@@ -0,0 +1,324 @@
+#include <iostream>
+
+#include <config.h>
+
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/ErrorMacros.hpp>
+#include <dune/common/timer.hh>
+#include <opm/simulators/linalg/bda/BlockedMatrix.hpp>
+
+#include <opm/simulators/linalg/bda/IncompleteSAI.hpp>
+
+namespace Opm
+{
+namespace Accelerator
+{
+    using Opm::OpmLog;
+    using Dune::Timer;
+
+    template <unsigned int block_size>
+    void IncompleteSAI<block_size>::setParams(int Nb_, int nnzbs_, int verbosity_){
+        this->Nb = Nb_;
+        this->bs = block_size;
+        this->N = bs * Nb_;
+        this->nnzbs = nnzbs_;
+        this->verbosity = verbosity_;
+    }
+
+    template <unsigned int block_size>
+    void IncompleteSAI<block_size>::buildMapping(int *CSRColIndices, int *CSRRowPointers){
+        Dune::Timer t_mapping;
+        std::vector<int> aux(CSRRowPointers, CSRRowPointers + Nb + 1);
```

Comment:
Should we make clear from the name what aux is for? Might help others to understand the code.

Reply:
Also added some comments.